src/main/java/com/minecolonies/coremod/client/gui/DialogDoneCancel.java

```diff
@@ -0,0 +1,192 @@
+package com.minecolonies.coremod.client.gui;
+
+import com.minecolonies.blockout.controls.Button;
+import com.minecolonies.blockout.controls.Label;
+import com.minecolonies.blockout.controls.Text;
+import com.minecolonies.blockout.Loader;
+import com.minecolonies.blockout.views.Window;
+import com.minecolonies.coremod.lib.Constants;
+import com.minecolonies.blockout.OverlayView;
+import com.minecolonies.coremod.util.Log;
+
+/**
+ * Manage windows and their events.
+ */
+public class DialogDoneCancel extends OverlayView implements Button.Handler
+{
+    /**
+     * buttonid when cancel is pressed.
+     */
+    public static final int CANCEL = 0;
+
+    /**
+     * buttonid when done is pressed.
+     */
+    public static final int DONE   = 1;
+
+    /**
+     * Resource suffix.
+     */
+    private static final String DIALOG_OK_CANCEL_SUFFIX = ":gui/dialogdonecancel.xml";
+
+    /**
+     * Label for the title of the dialog.
+     */
+    protected final Label   titleLabel;
+
+    /**
+     * Text for the text content of the dialog.
+     */
+    protected final Text    contentText;
+
+    /**
+     * Done button.
+     */
+    protected final Button  doneButton;
+
+    /**
+     * Cancel button.
+     */
+    protected final Button  cancelButton;
+
+    /**
+     * Handler for the onCloseDialog event.
+     */
+    protected       Handler handler;
+
+    /**
+     * Constructor for the DialogDoneCancel class.
+     *
+     * @param window in which the dialog will be displayed
+     */
+    public DialogDoneCancel(final Window window)
+    {
+        super();
+        Loader.createFromXMLFile(Constants.MOD_ID + DIALOG_OK_CANCEL_SUFFIX, this);
+        titleLabel = findPaneOfTypeByID("title", Label.class);
+        contentText = findPaneOfTypeByID("textcontent", Text.class);
+        doneButton = findPaneOfTypeByID("done", Button.class);
+        cancelButton = findPaneOfTypeByID("cancel", Button.class);
+        doneButton.setHandler(this);
+        cancelButton.setHandler(this);
+        this.window = window;
+        setPosition(0, 0);
+        setVisible(false);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "setVisible" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)